ListProj/Program.cs

```diff
@@ -0,0 +1,106 @@
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for new interns at Metamation, Batch - July 2023
+// Copyright (c) Metamation India.                                              
+// ------------------------------------------------------------------------
+// Program.cs
+// Implement a custom MyList<T> class using arrays as the underlying data structure.
+// The MyList<T> should start with an initial capacity of 4 and double its capacity when needed.
+// Use the template shown below for implementation. Throw exceptions when necessary. 
+// class MyList<T> {
+//   public MyList () { }
+//   public int Count { get; }
+//   public int Capacity { get; }
+//   public T this[int index] { get; set; }
+//   public void Add (T a) { }
+//   public bool Remove (T a) { }
+//   public void Clear () { }
+//   public void Insert (int index, T a) { }
+//   public void RemoveAt (int index) { }
+// }
+// IndexOutOfRangeException: This exception should be thrown when attempting to access an index that is out of the valid range. 
+// InvalidOperationException: This exception should be thrown when attempting to remove an item that is not found in the list. 
+// --------------------------------------------------------------------------------------------
+internal class Program {
+public static void Main () { }
+}
+#region MyList ------------------------------------------------------------------------------
+/// <summary>MyList class</summary>
+/// <typeparam name="T"></typeparam>
+public class MyList<T> {
+   private T[] mElements;
+   private int mCapacity, mCount;

```

Comment:
By default, a class, method or a field will be private. So, "private" keyword is not required here.